overlord/snapstate/storehelpers.go

```diff
@@ -20,31 +20,71 @@
 package snapstate
 
 import (
+	"sort"
+
 	"github.com/snapcore/snapd/overlord/auth"
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/store"
+	"github.com/snapcore/snapd/strutil"
 )
 
-func userFromUserID(st *state.State, userID int) (*auth.UserState, error) {
-	if userID == 0 {
-		return nil, nil
+func idForUser(user *auth.UserState) int {
+	if user == nil {
+		return 0
 	}
-	return auth.User(st, userID)
+	return user.ID
 }
 
-func updateInfo(st *state.State, snapst *SnapState, channel string, ignoreValidation bool, userID int) (*snap.Info, error) {
+func userIDForSnap(st *state.State, snapst *SnapState, fallbackUserID int) (int, error) {
+	userID := snapst.UserID
+	_, err := auth.User(st, userID)
+	if err == nil {
+		return userID, nil
+	}
+	if err != auth.ErrInvalidUser {
+		return 0, err
+	}
+	return fallbackUserID, nil
+}
+
+func userFromUserID(st *state.State, userIDs ...int) (*auth.UserState, error) {
```

Comment:
A doc string would be nice to understand why it can take multiple userIDs  but only returns a single UserState, i.e. that it returns the first valid user for the given IDs or something.